src/v/cloud_storage/recursive_directory_walker.cc

```diff
@@ -52,10 +52,12 @@ recursive_directory_walker::walk(
             ss::file target_dir = co_await open_directory(target);
             auto sub = target_dir.list_directory(
               [&files,
-               &current_cache_size,
-               &dirlist,
+               _current_cache_size = &current_cache_size,
+               _dirlist = &dirlist,
                _target{target},
                _tracker{tracker}](ss::directory_entry entry) -> ss::future<> {
+                  auto& current_cache_size{*_current_cache_size};
+                  auto& dirlist{*_dirlist};
```

Comment:
nit: should we use the new `ss::coroutine::lambda` thing here so we don't have to copy anything out of the capture list?

Reply:
for some reason ASan complained when I tried this, probably because lambda is called many times
but it worked fine with different approach (when the lambda is not anonymous but created on a stack so its lifetime is long enough)

Reply:
it requires that you co_await immediately on the statement including the lambda. so assigning it to the variable and co_awaiting on the variable won't work (it relies on temporary variable lifetime extension).